assets/javascripts/kitten/components/form/password-input/index.js

```diff
@@ -1,91 +1,89 @@
-import React, { PureComponent } from 'react'
+import React, { useState } from 'react'
 import styled from 'styled-components'
 import { TextInput } from '../../../components/form/text-input'
 import { PasswordIcon } from '../../../components/icons/password-icon'
 import COLORS from '../../../constants/colors-config'
 import PropTypes from 'prop-types'
 import { pxToRem } from '../../../helpers/utils/typography'
+import classNames from 'classnames'
 
 const StyledPasswordInput = styled.div`
   position: relative;
   display: flex;
-`
-
-const StyledTextInput = styled(TextInput)`
-  padding-right: ${pxToRem(40)};
-`
 
-const StyledIcon = styled.span`
-  display: flex;
-  position: absolute;
-  z-index: 1;
-  padding: 0 ${pxToRem(11)};
-  right: 0;
-  top: 0;
-  bottom: 0;
-  cursor: pointer;
+  .k-Form-PasswordInput__input {
+    padding-right: ${pxToRem(40)};
+  }
 
-  &[aria-pressed='true'] {
-    svg {
-      fill: ${COLORS.primary1};
-      transition: all 0.2s;
+  .k-Form-PasswordInput__icon {
+    display: flex;
+    position: absolute;
+    z-index: 1;
+    padding: 0 ${pxToRem(11)};
+    right: 0;
+    top: 0;
+    bottom: 0;
+    cursor: pointer;
+
+    &[aria-pressed='true'] {
+      svg {
+        fill: ${COLORS.primary1};
+        transition: all 0.2s;
+      }
     }
   }
 `
 
-export class PasswordInput extends PureComponent {
-  static propTypes = {
-    iconLabel: PropTypes.string.isRequired,
-    hiddenIconLabel: PropTypes.string.isRequired,
-    name: PropTypes.string,
-  }
-
-  static defaultProps = {
-    name: 'password',
-  }
+export const PasswordInput = ({
+  name,
+  iconLabel,
+  hiddenIconLabel,
+  ...others
+}) => {
 
-  constructor(props) {
-    super(props)
+  const [isHidden, setIsHidden ] = useState(true)
+  const type = isHidden ? 'password' : 'text'
+  const active = !isHidden
+  const iconTitle = isHidden ? iconLabel : hiddenIconLabel
 
-    this.state = {
-      isHidden: true,
-    }
+  const handleClick = () => {
+    setIsHidden(!isHidden)
   }
 
-  handleClick = () => {
-    this.setState({ isHidden: !this.state.isHidden })
-  }
-
-  handleKeyDown = event => {
+  const handleKeyDown = () => {
     const enterKeyCode = 13
 
-    if (event.keyCode == enterKeyCode) {
-      this.handleClick()
-    }
-  }
-
-  render() {
-    const { name, iconLabel, hiddenIconLabel, ...others } = this.props
-
-    const type = this.state.isHidden ? 'password' : 'text'
-
-    const active = !this.state.isHidden
+    if (event.keyCode == enterKeyCode) { handleClick }
+  } 
+
+  return (
+    <StyledPasswordInput className="k-Form-PasswordInput">
+      <TextInput
+        {...others}
+        name={name}
+        type={type}
+        className={classNames('k-Form-PasswordInput__input')}
+      />
+      <span
```

Comment:
Attention, pour des questions d'accessibilité, tout élément qui a un `onClick` doit être un `button`.

En tant que `button` tu n'auras plus besoin de `onKeyDown` ni de `tabIndex`. Pour reset les styles par défaut tu peux utiliser la classe utilitaire `k-u-reset-button`.

```suggestion
      <button
```